.github/workflows/_eng-tools-test.yaml

```diff
@@ -1,4 +1,4 @@
-name: eng/tools Test
+name: Templates - eng/tools - Test
```

Comment:
would it make sense to be more explicit and just call use prefix `template-`? I guess it will depend on how many more conventions we need to create.

Reply:
I was trying to keep the filenames short (and sort the templates to the top of the list), but I can use prefix `template-` instead of `_` if both @weshaggard and @konrad-jamrozik prefer this.

Reply:
Note that `template-` will sort in the middle of other templates in the file list.  And unfortunately I think template (aka "reusable") workflows need to be directly in the "workflows" directory (can't be in a subdirectory).

```
suppressions-test.yaml
template-eng-tools-test.yaml
typespec-validation-test.yaml
```

vs:

```
_eng-tools-test.yaml
suppressions-test.yaml
typespec-validation-test.yaml
```

Reply:
My main thing is to try and have a naming convention that tells us at a glance what things are so I would also be fine with something like `_template-` prefix.

Reply:
https://github.com/Azure/azure-rest-api-specs/pull/29169